mistletoe/markdown_renderer.py

```diff
@@ -0,0 +1,537 @@
+            token.children, max_line_length=max_child_line_length
+        )
+        return self.prefix_lines(
+            list(lines) or [""], token.leader + " ", " " * indentation
+        )
+
+    def render_table(
+        self, token: block_token.Table, max_line_length: int = None
+    ) -> Iterable[str]:
+        # note: column widths are not preserved; they are automatically adjusted to fit the contents.
+        content = [self.table_row_to_text(token.header), []]
+        content.extend(self.table_row_to_text(row) for row in token.children)
+        col_widths = self.calculate_table_column_widths(content)
+        content[1] = self.table_separator_line_to_text(col_widths, token.column_align)
+        return [
+            self.table_row_to_line(col_text, col_widths, token.column_align)
+            for col_text in content
+        ]
+
+    def render_thematic_break(
+        self, token: block_token.ThematicBreak, max_line_length: int = None
+    ) -> Iterable[str]:
+        return [token.line]
+
+    def render_html_block(
+        self, token: block_token.HTMLBlock, max_line_length: int = None
+    ) -> Iterable[str]:
+        return token.content.split("\n")
+
+    def render_link_reference_definition_block(
+        self, token: LinkReferenceDefinitionBlock, max_line_length: int = None
+    ) -> Iterable[str]:
+        # each link reference definition starts on a new line
+        for child in token.children:
+            yield from self.span_to_lines([child], max_line_length=max_line_length)
+
+    def render_blank_line(
+        self, token: BlankLine, max_line_length: int = None
+    ) -> Iterable[str]:
+        return [""]
+
+    # helper methods
+
+    def embed_span(
+        self,
+        leader: Fragment,
+        tokens: Iterable[span_token.SpanToken],
+        trailer: Fragment = None,
+    ) -> Iterable[Fragment]:
+        """
+        Makes fragments from `tokens` and embeds within a leader and a trailer.
+        The trailer defaults to the same as the leader.
+        """
+        yield leader
+        yield from self.make_fragments(tokens)
+        yield trailer or leader
+
+    def blocks_to_lines(
+        self, tokens: Iterable[block_token.BlockToken], max_line_length: int = None
+    ) -> Iterable[str]:
+        """
+        Renders a sequence of block tokens into a sequence of lines.
+        """
+        for token in tokens:
+            yield from self.render_map[token.__class__.__name__](
+                token, max_line_length=max_line_length
+            )
+
+    def span_to_lines(
+        self, tokens: Iterable[span_token.SpanToken], max_line_length: int = None
+    ) -> Iterable[str]:
+        """
+        Renders a sequence of span (inline) tokens into a sequence of lines.
+        """
+        fragments = self.make_fragments(tokens)
+        return self.fragments_to_lines(fragments, max_line_length=max_line_length)
+
+    def make_fragments(self, tokens: Iterable[span_token.SpanToken]):
```

Comment:
I.e. we could add this?

```suggestion
    def make_fragments(self, tokens: Iterable[span_token.SpanToken]
    ) -> Iterable[Fragment]:
```